content/docs/extension-author-guide/create-extension/why-dockerfiles.md

```diff
@@ -0,0 +1,105 @@
++++
+title="Why Dockerfiles"
+weight=402
++++
+
+<!-- test:suite=dockerfiles;weight=2 -->
+
+Let's see a build that requires base image extension in order to succeed.
+
+### Examine `hello-extensions` buildpack
+
+#### detect
+
+<!-- test:exec -->
+```bash
+cat $PWD/samples/buildpacks/hello-extensions/bin/detect
+```
+
+The buildpack always detects (because its exit code is `0`) but doesn't require any dependencies (as the output build plan is empty).
+
+#### build
+
+<!-- test:exec -->
+```bash
+cat $PWD/samples/buildpacks/hello-extensions/bin/build
+```
+
+The buildpack tries to use `tree` at build-time, and defines a launch process called `curl` that runs `curl --version` at runtime.
+
+### Create a builder with extensions and publish it
+
+Ensure you are authenticated with an OCI registry: `docker login` should succeed
+
+For test purposes, you can launch a local unauthenticated registry:
+
+<!-- test:exec -->
+```bash
+docker run -d --rm -p 5000:5000 registry:2
+```
+
+Set your preferred registry namespace (typically your username if using Docker Hub): `registry_namespace=<your preferred registry namespace>`
```

Comment:
As an end-user, if I've not come across this before I'm asking Is `registry_namespace\ an ENV var, an entry in a TOML file or a `pack` specific configuration.

Reply:
Updated in 235c3064 ...setting this variable is actually no longer helpful as we're always using localhost:5000 in the examples.